### DISCUSSION TITLE
Why does the gallery show images rotated/stretched?

### DISCUSSION THREAD

**Discussion by Scratch7951** [datenstrom/community#661] [Ask a question] [answered]
The Gallery extension is displaying some strange behaviour that I do not understand: wrongly rotated and/or stretched images in thumbnail and/or zoomed view.

It seems to occur when images have an EXIF Orientation flag other than 0. e.g. 1. That is, the image is rotated 90 degrees. This is "wrong" and I want to rotate the image so that it is oriented correctly when viewed with the Gallery extension. However, this is not going smoothly for me.

I have tried to correct the rotation in two ways. Firstly, using EXIFtool to set the orientation flag to something else (e.g. 0 or 1, depending on the source image).
`exiftool -Orientation=0 -n image.jpg`

Alternatively I have used Imagemagick to create a new rotated file:
`mogrify -rotate 90 image.jpg image_rotated.jpg`

In each case, the image looks perfectly fine in any previewer (e.g. Geeqie).

In Gallery, the thumbnail can look OK but the zoomed image can be incorrectly rotated and sometimes stretched and distorted. There seems to be no logic behind this and I have spent a lot of time trying work out the cause. No solution yet :(

Between each attempt, I have deleted the thumbnails and cleared the browser cache. I have even stopped the Web server and used `php yellow.php clean` in between making changes to the images. I feel that I am going around in circles so thought I should raise this is an issue in case it is, in fact a bug. Or (hopefully) I am just going something wrong.

Thanks for your help.

**Comment by markseuffert** [comment, marked as answer]
The [gallery extension](https://github.com/annaesvensson/yellow-gallery) uses [PhotoSwipe](https://github.com/dimsemenov/photoswipe) and what you describe appears to be a [known problem](https://github.com/dimsemenov/PhotoSwipe/search?q=exif&type=issues). As soon as EXIF orientation is supported in PhotoSwipe, it will become available in Datenstrom Yellow too. As a workaround remove the EXIF orientation from your JPG images or upload them with the web browser. When you upload large images with the web browser, they will be automatically reorientated. See also datenstrom/community#478.

**Comment by Scratch7951** [comment]
Thanks, @markseu . That workaround tip worked fine. I removed the EXIF Orientation flag (set it to zero) and uploaded the images as normal. Issue can be closed. :)
